src/app/auto-update-handler.ts

```diff
@@ -1,67 +1,119 @@
 
-    this.filename = '';
-    this.logFilePath = path.join(this.tempDir, 'auto_update.log');
+    if (this.autoUpdater) {
+      this.autoUpdater.logger = electronLog;
 
-    this.updateUtil = isDevEnv
-      ? path.join(
-          __dirname,
-          '../../../node_modules/auto-update/auto_update_helper.exe',
-        )
-      : path.join(path.dirname(app.getPath('exe')), 'auto_update_helper.exe');
+      this.autoUpdater.on('update-not-available', () => {
+        const mainWebContents = windowHandler.mainWebContents;
+        // Display client banner
+        if (mainWebContents && !mainWebContents.isDestroyed()) {
+          mainWebContents.send('display-client-banner', {
+            reason: 'autoUpdate',
+            action: 'update-not-available',
+          });
+        }
+      });
 
-    this.updateUtilArgs = [];
+      this.autoUpdater.on('update-downloaded', () => {
+        this.isUpdateAvailable = true;
+        const mainWebContents = windowHandler.mainWebContents;
+        // Display client banner
+        if (mainWebContents && !mainWebContents.isDestroyed()) {
+          mainWebContents.send('display-client-banner', {
+            reason: 'autoUpdate',
+            action: 'update-available',
+          });
+        }
+      });
+    }
   }
 
   /**
-   * Launch the auto update helper
+   * Installs the latest update quits and relaunches application
+   */
+  public updateAndRestart = async (): Promise<void> => {
+    if (!this.isUpdateAvailable) {
+      return;
+    }
+    // Handle update and restart for macOS
+    if (isMac) {
+      windowHandler.setIsAutoUpdating(true);
+    }
+    setImmediate(() => {
+      if (this.autoUpdater) {
+        this.autoUpdater.quitAndInstall();
+      }
+    });
+  };
+
+  /**
+   * Checks for the latest updates
+   * @return void
+   */
+  public checkUpdates = async (): Promise<void> => {
+    logger.info('auto-update-handler: Checking for updates');
+    if (this.autoUpdater) {
+      const updateCheckResult = await this.autoUpdater.checkForUpdatesAndNotify();
+      logger.info('auto-update-handler: ', updateCheckResult);
+    }
+    logger.info('auto-update-handler: After checking auto update');
+  };
+
+  /**
+   * Constructs the SDA auto update end point url
+   *
+   * @return string
+   * @example https://corporate.symphony.com/macos/general
    */
-  public async update(filename: string) {
-    logger.info(`auto-update-handler: Starting auto update!`);
-    this.filename = filename;
-    if (isWindowsOS) {
-      this.updateUtilArgs = [
-        this.filename,
-        app.getPath('exe'),
-        this.logFilePath,
-      ];
+  public getUpdateUrl = (): string => {
+    const { url: userConfigURL } = config.getUserConfigFields(['url']);
+    const { url: globalConfigURL } = config.getGlobalConfigFields(['url']);
+    const { autoUpdateUrl, autoUpdateChannel } = config.getConfigFields([
+      'autoUpdateChannel',
+      'autoUpdateUrl',
+    ]);
+
+    if (autoUpdateUrl) {
```

Comment:
Let's suppose autoUpdateUrl is a relative path: `/windows/general`
What would be the behaviour?

Reply:
auto updater will fail to fetch the updates and logs error. Should we have any validation in this case and direct it to using the channel instead?

Reply:
I think that option would be preferable yes!